hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx

```diff
@@ -1268,80 +1651,92 @@ cdef class CrossExchangeMarketMakingStrategy(StrategyBase):
                             f"Current hedging price: {effective_hedging_price:.8f} {market_pair.maker.quote_asset} "
                             f"(Rate adjusted: {effective_hedging_price_adjusted:.8f} {market_pair.taker.quote_asset})."
                         )
-                    order_id = self.c_place_order(market_pair, False, True, ask_size, ask_price)
+                    self.place_order(market_pair, False, True, ask_size, ask_price)
                 else:
-                    if self._logging_options & self.OPTION_LOG_NULL_ORDER_SIZE:
+                    if LogOption.NULL_ORDER_SIZE in self._logging_options:
                         self.log_with_clock(
                             logging.WARNING,
                             f"({market_pair.maker.trading_pair})"
                             f"Order book on taker is too thin to place order for size: {ask_size}"
                             f"Reduce order_size_portfolio_ratio_limit"
                         )
             else:
-                if self._logging_options & self.OPTION_LOG_NULL_ORDER_SIZE:
+                if LogOption.NULL_ORDER_SIZE in self._logging_options:
                     self.log_with_clock(
                         logging.WARNING,
                         f"({market_pair.maker.trading_pair}) Attempting to place a limit ask but the "
                         f"ask size is 0. Skipping. Check available balance."
                     )
 
-    cdef str c_place_order(self,
-                           object market_pair,
-                           bint is_buy,
-                           bint is_maker,  # True for maker order, False for taker order
-                           object amount,
-                           object price):
-        cdef:
-            str order_id
-            double expiration_seconds = NaN
-            object market_info = market_pair.maker if is_maker else market_pair.taker
-            object order_type = market_info.market.get_maker_order_type() if is_maker else \
-                market_info.market.get_taker_order_type()
+    def place_order(self,
+                    market_pair: MakerTakerMarketPair,
+                    is_buy: bool,
+                    is_maker: bool,  # True for maker order, False for taker order
+                    amount: Decimal,
+                    price: Decimal,
+                    maker_order_id: str = None,
+                    maker_exchange_trade_id: str = None):
+        expiration_seconds = s_float_nan
+        market_info = market_pair.maker if is_maker else market_pair.taker
+        order_type = market_info.market.get_maker_order_type() if is_maker else \
+            market_info.market.get_taker_order_type()
         if order_type is OrderType.MARKET:
             price = s_decimal_nan
```

Comment:
I still strongly believe that when we want to create a taker order, we should do that using Limit orders and not Market orders. And I think we agreed on this already in one of the discussions post stand up meeting we had about the XEMM adaptations. Please @mhrvth review this, because it seems that we will use Market orders if the connector we are using returns that for taker orders.

Reply:
✅